### PR TITLE
191 Site Edition

### PR DESCRIPTION
**Pull request by theo-auffeuvre**
Possibility to edit a site

### CODE REVIEW

File: canopeum_frontend/src/components/analytics/OptionQuantitySelector.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ const OptionQuantitySelector = <TValue extends OptionQuantityValueType>(
 
   useEffect(() => setAvailableOptions(options), [options])
   useEffect(() => setFilteredOptions(availableOptions), [availableOptions])
-  useEffect(() => onChange(selectedOptions), [selectedOptions, onChange])
+  useEffect(() => onChange(selectedOptions), [selectedOptions])
```
Comment (Samuel-Therrien-Beslogic) [inline canopeum_frontend/src/components/analytics/OptionQuantitySelector.tsx:41]: If you need to remove this, please comment why we're ignoring this warning

File: canopeum_frontend/src/components/analytics/TreeSpeciesSelector.tsx [hunk 2]
```diff
@@ -38,36 +38,41 @@ const TreeSpeciesSelector = (
...
+      return {
+        option: {
+          displayText: matchingSpecie
+            ? translateValue(matchingSpecie)
+            : '',
+          value: specie.id,
+        },
+        quantity: specie.quantity,
+      }
+    })), [availableSpecies])
```
Comment (Samuel-Therrien-Beslogic) [inline canopeum_frontend/src/components/analytics/TreeSpeciesSelector.tsx:55]: Similar comment about indicating why we *need* to omit useEffect dependencies

File: canopeum_backend/canopeum_backend/serializers.py [hunk 3]
```diff
@@ -255,10 +255,14 @@ class Meta:
...
     fr = serializers.SerializerMethodField()
+    type_id = serializers.SerializerMethodField()
 
     class Meta:
         model = Sitetreespecies
-        fields = ("id", "quantity", "en", "fr")
+        fields = ("id", "type_id", "quantity", "en", "fr")
+
+    def get_type_id(self, obj) -> int:
+        return TreeTypeSerializer(obj.tree_type).data.get("id", None)  # type: ignore[no-any-return]
```
Comment (Samuel-Therrien-Beslogic) [inline canopeum_backend/canopeum_backend/serializers.py:265]: Yeah not much we can do here, `data` is a `dict[Unknown, Unknown]` (`dict[Any, Any] in mypy`). Even if you explicitly type the `obj`

File: canopeum_backend/canopeum_backend/views.py [hunk 4]
```diff
@@ -347,14 +348,16 @@ def patch(self, request: Request, siteId):
         serializer = SiteSerializer(site, data=request.data, partial=True)
         if serializer.is_valid():
             site = serializer.save(
-                image=image,
+                image=image if request.data.get("image") is not None else site.image,  # pyright: ignore[reportPossiblyUnboundVariable]
```
Comment (Samuel-Therrien-Beslogic) [inline canopeum_backend/canopeum_backend/views.py:351]: You should instead first assign `image = site.image` in the else path of `if request.data.get("image") is not None:`

Note that we should probably eventually differenciate between "do nothing" (no image param) and "remove image" (image param set to None ?)

```suggestion
                image=image,
```

File: canopeum_frontend/src/components/analytics/OptionQuantitySelector.tsx (hunk 1 above)
Comment (Samuel-Therrien-Beslogic) [inline canopeum_frontend/src/components/analytics/OptionQuantitySelector.tsx:41]: If you need to remove this, please comment why we're ignoring this warning

File: canopeum_frontend/src/components/analytics/TreeSpeciesSelector.tsx (hunk 2 above)
Comment (Samuel-Therrien-Beslogic) [inline canopeum_frontend/src/components/analytics/TreeSpeciesSelector.tsx:55]: Similar comment about indicating why we *need* to omit useEffect dependencies

**Review by Samuel-Therrien-Beslogic — CHANGES_REQUESTED** [review]

File: canopeum_frontend/src/constants/weatherImageMap.ts [hunk 5]
```diff
@@ -1,24 +1,21 @@
...
-
 const WMOCategoriesImages: WMOCategoriesImageUrls[] = [
   {
-    imageUrl: '@assets/images/weather/sunny-bg.jpeg',
+    imageUrl: readonlyURL('../assets/images/weather/sunny-bg.jpeg', import.meta.url),
     WMOCategories: ['Clear sky', 'Mainly clear'],
   },
   {
-    imageUrl: getURLString('@assets/images/weather/cloudy-bg.jpeg'),
+    imageUrl: readonlyURL('../assets/images/weather/cloudy-bg.jpeg', import.meta.url),
```
Comment (Samuel-Therrien-Beslogic) [inline canopeum_frontend/src/constants/weatherImageMap.ts:14]: Can we no use webpack/ts-paths style `@` isntead of relative paths ?

File: canopeum_frontend/src/constants/weatherImageMap.ts [hunk 6]
```diff
@@ -62,9 +59,11 @@ const getImageNameByWMOCategories = (WMOCategory: string): string | undefined =>
...
     ) ?? undefined
 
-  return image
-    ? image.imageUrl
-    : WMOCategoriesImages[0].imageUrl
+  return image && image.imageUrl
+    ? image.imageUrl.href
+    // image is defined manually, so it is guaranteed to have an imageUrl
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- comment above
+    : WMOCategoriesImages[0].imageUrl!.href
```
Comment (Samuel-Therrien-Beslogic) [inline canopeum_frontend/src/constants/weatherImageMap.ts:66]: The return type is already possibly undefined

```suggestion
    : WMOCategoriesImages[0].imageUrl?.href
```

**Review by Samuel-Therrien-Beslogic — CHANGES_REQUESTED** [review]

File: canopeum_frontend/src/constants/weatherImageMap.ts (hunk 5 above)
Comment (theo-auffeuvre) [reply canopeum_frontend/src/constants/weatherImageMap.ts:14]: Unfortunaly it didn;t with relative path :(

**Pull request merged**